### ISSUE TITLE
"az bot create" provides no way to specify pre-existing QnAmaker, Web App, Service plan etc ...

### ISSUE THREAD

**Issue by archmangler** [Azure/azure-cli#11433]
**Describe the bug**

Incomplete azure bot cli utility:

- There is no documented way to create an azure bot via cli (without the use of ARM template) and associate it with an existing QnAmaker and webapp service.

We can create the dependent services like qnamaker and web app service for the azure bot service using az cli, but there is no way to create a bot and specify these dependency services at creation time, or even using the "az bot update" command.

**To Reproduce**

- Step 1: create all the foundational services for the azure bot using these commands:

```
# Create resource group.
az group create --name devexp-qnamaker --location westus

# Create Search instance and get admin key.
# in this example, the key is "B044183F2450193E26751DC20423DFE4"
az search service create -g devexp-qnamaker -n devexp-qnamaker-search --sku standard
az search admin-key show -g devexp-qnamaker --service-name devexp-qnamaker-search

# Create App Service plan and Web App
# in this example, Web App endpoint will be https://devexp-qnamaker-webapp.azurewebsites.net
az appservice plan create -g devexp-qnamaker -n devexp-qnamaker-webapp-plan --sku S1
az webapp create -g devexp-qnamaker -n devexp-qnamaker-webapp --plan devexp-qnamaker-webapp-plan

# Configure Web App
az webapp config appsettings set -g devexp-qnamaker -n devexp-qnamaker-webapp --settings \
    AzureSearchName=devexp-qnamaker-search \
    AzureSearchAdminKey=B044183F2450193E26751DC20423DFE4 \
    PrimaryEndpointKey=devexp-qnamaker-webapp-PrimaryEndpointKey \
    SecondaryEndpointKey=devexp-qnamaker-webapp-SecondaryEndpointKey \
    DefaultAnswer="No good match found in KB." \
    QNAMAKER_EXTENSION_VERSION="latest"
az webapp cors add -g devexp-qnamaker -n devexp-qnamaker-webapp -a "*"

# Create QnAMaker account
az cognitiveservices account create -g devexp-qnamaker -n devexp-qnamaker-account --kind QnAMaker --sku S0 --location westus --api-properties qnaRuntimeEndpoint=https://devexp-qnamaker-webapp.azurewebsites.net
```

- Step 2: Refer to the az bot  documentation to find how to create a bot an associate it with the pre-create QnAMaker service, service plan and Web App:

```
https://docs.microsoft.com/en-us/cli/azure/bot?view=azure-cli-latest#az-bot-create
```

Result: Nothing available in the documentation.

Impact: 

No means to completely script bot creation using these elements without resorting to the portal. This means that for our large scale enterprise requirements we cannot smoothly automate bot creation via ci/cd pipelines at speed without substantial manual work in the portal. Due to the high rate of iteration required to develop bots to meet requirements, this is a blocker.

**Expected behavior**

- There should be a complete documented example of how to instantiate an azure chat bot and associate it with the QnAmaker and Web App created using the command sequence above.
- This allows us to script QnAmaker and associated services with custom configuration and associate new chatbots with these components.

**Environment summary**

- All environments

**Additional context**

- We are avoiding the use of ARM templates to automate this deployment since they are overly complex, confusing, with gaps in documentation and poor examples.

**Comment by archmangler** [comment]
Any feedback ?

**Comment by stevengum** [comment]
@archmangler you should use `az webapp config appsettings` to configure the App Service hosting your bot.

`az bot update` affects the Bot Registration, whether it is a Web App Bot or a Bot Channels Registration. It does not affect the settings on the App Service.

**Comment by archmangler** [comment]
@stevengum  - would you be able to provide an example please?

**Comment by stevengum** [comment]
Instead of the commands you used above, you would do the following:

### Create App Service plan and Web App for hosting the bot:
```bash
az appservice plan create -g devexp-bot-n devexp-bot-webapp-plan --sku S1
az webapp create -g devexp-bot -n devexp-bot-webapp --plan devexp-bot-webapp-plan
```

As an example we'll use [these keys](https://github.com/microsoft/BotBuilder-Samples/blob/master/samples/javascript_nodejs/11.qnamaker/.env#L3) from the QnAMaker sample. You should use the keys that your bot/app looks for in the Application Settings.

#### Configure Web App:
```bash
az webapp config appsettings set -g devexp-bot -n devexp-bot-webapp --settings \
    QnAKnowledgebaseId=devexp-qnamaker-knowledgebaseid \
    QnAEndpointKey=devexp-qnamaker-webapp-PrimaryEndpointKey \
    QnAEndpointHostName="https://devexp-qnamaker-webapp.azurewebsites.net" \
```

**Comment by stevengum** [comment]
Closing issue. If the provided answer does not solve your ask, please @-mention me in this issue.

**Comment by archmangler** [comment]
Hi @stevengum I've tested the command sequence you provided earlier. This dies not result in an actual chatbot. It does result in the following resources being created in my resource group:

1. A service plan
2. an App service
3. Cognitive search service

But no resource of type "Web app Bot". To obtain the web app bot I have to run the command:

```
az bot create --resource-group "rsg-whatever" \
 --appid  "xxxxwhatever" \
 --display-name "Support-Bot" \
 --kind webapp \
 --name \"supportbot\" \
 --password "somepasword" \
 --lang "Csharp" \
 --sku "S1"
```

This creates a resource of type "Web App Bot". However, it also creates along with it the following two new resources:

1. An "App Service Plan"
2. An "App Service"

This is a problem because I did not specify these resources and would like to configure the chat bot to use the separately create app service plan and service. However, as mentioned before looking at the manual for "az bot create" and "az bot update" there is nothing to allow one to specify pre-existing app service or app service plans.
